frontends/web/src/components/devices/bitbox02bootloader/toggleshowfirmwarehash.tsx

```diff
@@ -28,33 +28,36 @@ interface LoadedProps {
     enabled: boolean;
 }
 
-type Props = ToggleProps & LoadedProps & TranslateProps;
+type Props = ToggleProps & LoadedProps;
 
-class ToggleFWHash extends Component<Props, {}> {
-    private handleToggle = event => {
+const ToggleFWHash: FunctionComponent<Props> = ({ enabled, deviceID }) => {
+    const { t } = useTranslation();
+    const [enabledState, setEnabledState] = useState<boolean | undefined>();
+
+    const check = enabledState !== undefined ? enabledState : enabled;
+
+    const handleToggle = event => {
+        const enabled: boolean = event.target.checked;
         apiPost(
-            'devices/bitbox02-bootloader/' + this.props.deviceID + '/set-firmware-hash-enabled',
-            event.target.checked,
+            'devices/bitbox02-bootloader/' + deviceID + '/set-firmware-hash-enabled',
+            enabled,
         );
+        setEnabledState(enabled)
     }
 
-    public render() {
-        const { t, enabled } = this.props;
-        return (
-            <div className="box slim divide">
-                <div className="flex flex-row flex-between flex-items-center">
-                    <p className="m-none">{t('bb02Bootloader.advanced.toggleShowFirmwareHash')}</p>
-                    <Toggle
-                        checked={enabled}
-                        id="togggle-show-firmware-hash"
-                        onChange={this.handleToggle}
-                        className="text-medium" />
-                </div>
+    return (
+        <div className="box slim divide">
+            <div className="flex flex-row flex-between flex-items-center">
+                <p className="m-none">{t('bb02Bootloader.advanced.toggleShowFirmwareHash')}</p>
+                <Toggle
+                    checked={check}
+                    id="togggle-show-firmware-hash"
+                    onChange={handleToggle}
+                    className="text-medium" />
             </div>
-        );
-    }
+        </div>
+    );
 }
 
-const loadHOC = load<LoadedProps, ToggleProps & TranslateProps>(({ deviceID }) => ({ enabled: 'devices/bitbox02-bootloader/' + deviceID + '/show-firmware-hash-enabled' }))(ToggleFWHash);
-const HOC = translate()(loadHOC);
+const HOC = load<LoadedProps, ToggleProps>(({ deviceID }) => ({ enabled: 'devices/bitbox02-bootloader/' + deviceID + '/show-firmware-hash-enabled' }))(ToggleFWHash);
```

Comment:
Very nice with the hooks, once this is merged I'll will look into typing the api for `devices/bitbox02-bootloader/`